godot-core/src/registry.rs

```diff
@@ -204,6 +204,13 @@ fn fill_class_info(component: PluginComponent, c: &mut ClassRegistrationInfo) {
             fill_into(
                 &mut c.godot_params.create_instance_func,
                 generated_create_fn,
+            )
+            .unwrap_or_else(|_|
+                panic!(
+                    "the definition for `{}` is overloaded; you may have multiple `GodotClass`es named `{}` which you should #[class(rename=...)]",
```

Comment:
"overloaded" is a bit imprecise here, but I also think there is no need to repeat the class name.

Maybe simply:
```suggestion
                    "class `{}` is already defined; to register equally named structs under different names, use `#[class(rename=...)]`",
```

Reply:
I think it would be good to keep `GodotClass` in there to point the user to that symbol (I guess `class` is technically the relevant symbol, but it reads as generic here).

I'll take another shot at this error message

Reply:
This would still need to be updated 🙂


Regarding `GodotClass`, you could mention:
```rs
                    "class `{}` is already defined; to register equally named structs under\n\
                    different names, use `#[derive(GodotClass)` with `#[class(rename=...)]`"
```
(it might deserve the newline)